src/apps/polynomial/optimism/polynomial.balance-fetcher.ts

```diff
@@ -0,0 +1,51 @@
+import { Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { presentBalanceFetcherResponse } from '~app-toolkit/helpers/presentation/balance-fetcher-response.present';
+import { BalanceFetcher } from '~balance/balance-fetcher.interface';
+import { Network } from '~types/network.interface';
+
+import { PolynomialContractFactory, PolynomialCoveredCall } from '../contracts';
+import { POLYNOMIAL_DEFINITION } from '../polynomial.definition';
+
+const network = Network.OPTIMISM_MAINNET;
+
+@Register.BalanceFetcher(POLYNOMIAL_DEFINITION.id, network)
+export class OptimismPolynomialBalanceFetcher implements BalanceFetcher {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(PolynomialContractFactory) private readonly contractFactory: PolynomialContractFactory,
+  ) {}
+
+  async getVaultBalances(address: string) {
+    return this.appToolkit.helpers.masterChefContractPositionBalanceHelper.getBalances<PolynomialCoveredCall>({
```

Comment:
You're better off using `this.appToolkit.helpers.contractPositionBalanceHelper` in this case, since this isn't a MasterChef contract.